rfhub2/cli/api_client.py

```diff
@@ -0,0 +1,79 @@
+from requests import session, post, get, delete, exceptions
+from typing import Dict, List
+
+from rfhub2.model import Collection, Keyword
+
+
+API_V1 = 'api/v1'
+TEST_COLLECTION = {"name": "healtcheck_collection", "type": "a", "version": "a", "scope": "a",
+                   "named_args": "a", "path": "a", "doc": "a", "doc_format": "a"}
+
+
+class Client(object):
+    """
+    API client with methods to populate rfhub2 application.
+    """
+    def __init__(self, app_url: str, user: str, password: str):
+        self.app_url = app_url
+        self.session = session()
+        self.api_url = f'{self.app_url}/{API_V1}'
+        self.session.auth = (user, password)
+        self.session.headers = {"Content-Type": "application/json", "accept": "application/json"}
+
+    def check_communication_with_app(self) -> None:
+        try:
+            self.get_collections()
+        except exceptions.RequestException:
+            print(f'Connection to application at {self.app_url} refused!\n'
+                  f'Check parameters for app-interace and port.')
+            exit(1)
+        req_check_auth = self._post_request('collections', TEST_COLLECTION)
+        if req_check_auth['name'] == 'healtcheck_collection':
+            self._delete_request('collections', req_check_auth["id"])
+        else:
+            print(f'{req_check_auth.reason}! Check used credentials.')
+            exit(1)
+
+    def get_collections(self) -> List[Collection]:
```

Comment:
@pbylicki Should I check this methods similar to this test:
https://github.com/pbylicki/rfhub-new/blob/master/tests/api/endpoints/healthcheck_tests.py#L17
Or should I start app similarly to BaseEnpointTests and connect to working app?

Reply:
I would mock requests library methods (get, post, etc.) that are used to perform actual http request with different responses that you need for each scenario. You can either prepare the mocking by yourself or use some utility like this: https://github.com/getsentry/responses
I you want I can help you with some initial test as an example.

Reply:
> I you want I can help you with some initial test as an example.
 
That would be very good idea. I tried mockng and responses, and dependency override and got stuck.

Reply:
Yes, this is what I meant. Note that in `rsps.add` you can just pass `json=RESP` to provide expected body.

Reply:
That's great! I've aded some test for api_client.py. Coverage is 100% with more mocks than real code ;)